### DISCUSSION TITLE
Adjust TOC highlight logic

### DISCUSSION THREAD

**Discussion by ghost** [withastro/starlight#2542] [Feature Requests]
### What version of `starlight` are you using?

0.28.4

### What is your idea?

Support custom TOC highlighting logic.

### Why is this feature necessary?

First of all, I think starlight is very good.
But I think the TOC logic can be customized.
For example, I hope that when I reach the bottom of the article, the last title can be highlighted.
Of course, there are also many people who like the current mode.
So, can we add a configuration item to support custom TOC highlighting logic?

### Participation

- [ ] I am willing to submit a pull request for this feature.

**Comment by trueberryless** [comment]
I just wanna make sure that Chris' (@delucis) thoughts are persistently stored here as well ([source](https://discord.com/channels/830184174198718474/1070481941863878697/1301190367286136959)):

> One pattern I've seen recently in some docs that would solve this is not just to highlight a single item but all on-screen items. I actually kind of like that, but would probably need some design changes.
